jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/PathMappingsHandler.java

```diff
@@ -111,11 +113,48 @@ public boolean handle(Request request, Response response, Callback callback) thr
             return false;
         }
         Handler handler = matchedResource.getResource();
+        PathSpec pathSpec = matchedResource.getPathSpec();
         if (LOG.isDebugEnabled())
             LOG.debug("Matched {} to {} -> {}", pathInContext, matchedResource.getPathSpec(), handler);
-        boolean handled = handler.handle(request, response, callback);
+
+        PathSpecRequest pathSpecRequest = new PathSpecRequest(request, pathSpec);
+        boolean handled = handler.handle(pathSpecRequest, response, callback);
```

Comment:
Should we make this optional?  Perhaps not all usages will want the context set... specially if the handler mapped is itself a ContextHandler.   Or maybe this is in a PathMappingsHandler.Contextual   subclass?